File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.py
def solution(lines):
    a = 0
    answer = set()  # 중복 방지를 위해 set 사용
    box = []  # 지나간 구간을 저장하는 리스트

    for i in lines:
        for z in range(i[0], i[1]):  # 선분의 끝값(i[1]) 포함 X
            if z in box:
                answer.add(z)  # 두 개 이상의 선분이 겹치는 구간을 저장
            box.append(z)  # 지나간 구간 저장

    if len(answer) > 0:  # 겹치는 구간이 있다면
        a += len(answer)  # 겹치는 길이 계산

    return a

File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.java
import java.security.KeyStore.Entry;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;
import java.util.stream.Collector;
import java.util.stream.Collectors;
import java.util.stream.IntStream;
import java.util.stream.Stream;
import java.util.Scanner;
class Solution {
    public int solution(int[][] lines) {
    int answer = 0;
    Map <Integer, Integer> counter = new HashMap<>();
    for (int[] i : lines){
        for (int j = i[0]; j < i[1]; j++){
            counter.put(j, counter.getOrDefault(j, 0) + 1);
        }    
    }

    for (Integer i : counter.values()){
        if (i >= 2){
            answer += 1;
        }
    }
    return answer;
}
}